influencetx/core/tests/test_utils.py

```diff
@@ -42,3 +82,78 @@ def test_non_party_input(self):
         with mock.patch.object(utils, 'log') as log:
             assert utils.party_enum('not a real party') is None
         log.warn.assert_called_once_with("Could not determine party from 'not a real party'")
+
+
+class TestPartyLabel():
+
+    def test_null_input(self):
+        utils.party_label(None) == constants.PARTY_LABELS[constants.Party.UNKNOWN]
+
+    def test_invalid_input(self):
+        with assert_log_warning():
+            assert utils.party_label('not valid') == ''
+
+    def test_democrat_label(self):
+        utils.party_label('democrat') == constants.PARTY_LABELS[constants.Party.DEMOCRATIC]
+
+    def test_independent_label(self):
+        utils.party_label('independent') == constants.PARTY_LABELS[constants.Party.INDEPENDENT]
+
+    def test_republican_label(self):
+        utils.party_label('republican') == constants.PARTY_LABELS[constants.Party.REPUBLICAN]
+
+
+class TestHandleError():
+
+    def setup(self):
+        self.mock_handler = mock.Mock()
+        self.error = ValueError()
+
+    def test_defaults(self):
+        @utils.handle_error(ValueError, self.mock_handler)
+        def function_that_raises_value_error():
+            raise self.error
+
+        with assert_handle_error_logs():
+            function_that_raises_value_error()
+        self.mock_handler.assert_called_once_with(self.error)
+
+    def test_log_warning(self):
+        @utils.handle_error(ValueError, self.mock_handler, log_level='warn')
+        def function_that_raises_value_error():
+            raise self.error
+
+        with assert_handle_error_logs(log_level='warn'):
+            function_that_raises_value_error()
+
+    def test_not_logged(self):
+        @utils.handle_error(ValueError, self.mock_handler, log_level=None)
+        def function_that_raises_value_error():
+            raise self.error
+
+        with mock_handle_error_logging() as mock_logging:
+            function_that_raises_value_error()
+            mock_logging.getLogger.assert_not_called()
+
+
+def assert_log_warning(message=None):
```

Comment:
The file "influencetx/core/testing.py" also contains a "def assert_log_warning".  Would these collide with one another?

Reply:
Good question: No, as you can see, this function actually calls the one in `testing.py`. The `testing` module is imported in this file (`from influencetx.core import testing`) and `assert_log_warning` is called an attribute to that module. If the `testing` module were imported as `from influencetx.core.testing import *` then the naming would collide because that import syntax brings in all the attributes of the imported module into the scope of the module where the import is invoked.